### ISSUE TITLE
Edit recently sent messages

### ISSUE THREAD

**Issue by AlexHalb** [eternagame/HTML-Chat#92]
It would be nice to have a feature that would allow a user to edit a message they just sent. I think a lot of people would use it to fix spelling and grammar errors they notice after they send a message. This would probably be similar to Slack and Discord where the up arrow key triggers the edit. This might be difficult to implement with our current framework, so it may not be plausible. I figured I might as well through it out there and see if it gets implemented.

**Comment by luxaritas** [comment]
This is not yet implemented in the IRC protocol. There is a WIP spec: https://github.com/ircv3/ircv3-specifications/pull/304 The latest comment there says that they are going to be creating a new version of that draft spec based on https://gist.github.com/jesopo/ef44e928aeb1c893e722909608960b1b

This will need to get supported by txircd before we can implement it in the client. We can of course create our own implementation before it gets ratified (and that can help in the process of getting it ratified), though it might be good to wait until the new version of the spec is available.

A couple related issues:
- Users should be able to delete their own messages
- Ops should be able to delete any messages (this one I care about the most given that it can help us clean out spam or bot posts or whatnot, it’s obnoxious that we have to leave them there)
- Not sure if this is addressed in the spec, but edits and deletes should apply to chat history
- It would be good if ops could track edits/deletes to monitor for potential abuse (Eg, posting something inappropriate and then removing it)
